src/FileDownloader.ts

```diff
@@ -30,6 +31,47 @@ export default class FileDownloader implements IFileDownloader {
         return path.join(context.globalStorageUri.fsPath, `file-downloader-downloads`);
     }
 
+    /**
+     * @param settings optional additional settings for the download. If this is not provided, we will set the Accept header
+     * to application/octet-stream.
+     */
+    public async downloadFileFromGitHubRelease(
+        owner: string,
+        repository: string,
+        fileName: string,
+        context: ExtensionContext,
+        cancellationToken?: CancellationToken,
+        onDownloadProgressChange?: (downloadedBytes: number, totalBytes: number | undefined) => void,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        settings = { headers: { Accept: `application/octet-stream` } } as FileDownloadSettings
+    ): Promise<Uri> {
+        const url = await this.getGitHubDownloadLink(owner, repository, fileName);
+        if (url === undefined) {
+            throw new Error(`Failed to get download link for ${fileName}.`);
+        }
+
+        // Make sure we always have the Accept header set to application/octet-stream
+        if (settings.headers === undefined) {
+            settings.headers = {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                Accept: `application/octet-stream`
+            };
+        }
+        else if (settings.headers.Accept === undefined) {
+            settings.headers.Accept = `application/octet-stream`;
+        }
+
+        // If the GITHUB_TOKEN is set, then use it to download the file
+        // and then we wont get rate limited.
+        const token = process.env.GITHUB_TOKEN;
```

Comment:
💡 Just flagging if we know of any misuse of `GH token` here, no logging is happening in the underlying nested funct? (which could cause GH Token to get logged, and could be sniffed that way). If this is all hooked to already existing mechanism then we can take a look latter for this.  Also, purely for awareness and viz - @hsubramanianaks

Reply:
The token held in the GITHUB_TOKEN is scoped to this repository and expires after the job completes.

https://docs.github.com/en/actions/security-guides/automatic-token-authentication